### PR TITLE
Update release workflow (again)

### PR DESCRIPTION
**Pull request by maread99**
Attempts to f!x install and import checks on release workflow (#207).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -30,12 +30,22 @@ jobs:
...
-        # pip install --extra-index-url https://test.pypi.org/simple exchange_calendars==${{ github.ref_name }}
-        # python -c 'import exchange_calendars;print(exchange_calendars.__version__)'
-        # pip uninstall -y exchange_calendars
+        pip install --index-url https://test.pypi.org/simple exchange_calendars==${{ github.ref_name }} --no-deps
+
+    - name: Import test release
+      run: |
+        # install local package to download dependencies from pypi (#207)
+        pip install .
+        mv exchange_calendars exchange_calendars_copy
```
Comment (gerrymanoim) [inline .github/workflows/release.yml:42]: Why is this `mv` required?

**Comment by gerrymanoim** [comment]
Just a single question but LGTM (thanks for grabbing, sorry I've been traveling ).

File: .github/workflows/release.yml (hunk 1 above)
Comment (maread99) [reply .github/workflows/release.yml:42]: Hi Gerry, I'm ditching this bit.

**Comment by maread99** [comment]
I think I might be finally there with this. I've got pretty much the same workflow seemingly working on [market_prices](https://github.com/maread99/market_prices/blob/0.9.9/.github/workflows/release.yml).

For the history behind why it is as it is, see #207 and https://github.com/maread99/market_prices/issues/22, plus links from those conversations.

For whatever reason, simply waiting 10 seconds between publishing and installing doesn't cut the mustard, whilst enquiring until you get confirmation that the version is available seems to never fail - although this always take less than 10 seconds 🤷‍♂️.

What also appears important is to avoid requesting the package from the main PyPI repo during the publishing to / installation from the test repo - it seemed to cache knowledge that the new version wasn't available from the main repo and then not look again after it was actually published. This workflow avoids this possibility by, during the test repo stage, getting pip to only look at the test repo for `exchange_calendars` (via --index-url) and then installing the dependencies separately via `requirements.txt`.

Anyway, I think (🤞) this works.

**Comment by gerrymanoim** [comment]
Go for it!

**Pull request merged**